addons/account_edi/models/account_edi_document.py

```diff
@@ -224,7 +224,7 @@ def _process_documents_web_services(self, job_count=None, with_commit=True):
             move_to_lock = documents.move_id
             attachments_potential_unlink = documents.sudo().attachment_id.filtered(lambda a: not a.res_model and not a.res_id)
             try:
-                with self.env.cr.savepoint(flush=False):
+                with self.env.cr.savepoint(flush=False): # TODO Savepoint in a loop
```

Comment:
@smetl see above

Reply:
I agree. `SKIP LOCKED` seems indeed a good idea to detect the already locked records without creating a new savepoint.

Reply:
I tried to use `SKIP LOCKED` for something unrelated, but it was still raising in some cases if the `SKIP LOCKED` was not the very first thing in the transaction (because it got locked in the meantime) :thinking: